device/iot-device-samples/multiplexing-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/MultiplexingSample.java

```diff
@@ -0,0 +1,169 @@
+                    "Expected 3 arguments but received: %d.\n"
+                            + "The program should be called with the following args: \n"
+                            + "1. [Device 1 connection string] - String containing Hostname, Device Id & Device Key in one of the following formats: HostName=<iothub_host_name>;DeviceId=<device_id>;SharedAccessKey=<device_key>\n"
+                            + "2. [Device 2 connection string] - String containing Hostname, Device Id & Device Key in one of the following formats: HostName=<iothub_host_name>;DeviceId=<device_id>;SharedAccessKey=<device_key>\n"
+                            + "3. [Device 3 connection string] - String containing Hostname, Device Id & Device Key in one of the following formats: HostName=<iothub_host_name>;DeviceId=<device_id>;SharedAccessKey=<device_key>\n"
+                            + "4. [Protocol]                   - amqps | amqps_ws\n",
+                    args.length);
+            return;
+        }
+
+        String connString1 = args[0];
+        String connString2 = args[1];
+        String connString3 = args[2];
+        IotHubClientProtocol protocol = IotHubClientProtocol.AMQPS;
+        if (args[3].equalsIgnoreCase("amqps_ws"))
+        {
+            protocol = IotHubClientProtocol.AMQPS_WS;
+        }
+
+        MultiplexingClient multiplexingClient = new MultiplexingClient(protocol);
+        DeviceClient[] multiplexedDeviceClients = new DeviceClient[MULTIPLEXED_DEVICE_CLIENT_COUNT];
+        MultiplexedDeviceConnectionStatusChangeTracker[] connectionStatusTrackers = new MultiplexedDeviceConnectionStatusChangeTracker[MULTIPLEXED_DEVICE_CLIENT_COUNT];
+
+        for (int i = 0; i < MULTIPLEXED_DEVICE_CLIENT_COUNT; i++)
+        {
+            multiplexedDeviceClients[i] = new DeviceClient(args[i], protocol);
+            String deviceId = multiplexedDeviceClients[i].getConfig().getDeviceId();
+            connectionStatusTrackers[i] = new MultiplexedDeviceConnectionStatusChangeTracker();
+            multiplexedDeviceClients[i].registerConnectionStatusChangeCallback(connectionStatusTrackers[i], deviceId);
+            multiplexingClient.registerDeviceClient(multiplexedDeviceClients[i]);
+        }
+
+        System.out.println("Opening multiplexed connection");
+        // All previously registered device clients will be opened alongside this multiplexing client
+        multiplexingClient.open();
+        System.out.println("Multiplexed connection opened successfully");
+
+        for (int i = 0; i < MULTIPLEXED_DEVICE_CLIENT_COUNT; i++)
+        {
+            Message message = new Message("some payload");
+            multiplexedDeviceClients[i].sendEventAsync(message, new EventCallback(), message.getMessageId());
+        }
+
+        System.out.println("Waiting while messages get sent asynchronously...");
+        while (acknowledgedSentMessages < MULTIPLEXED_DEVICE_CLIENT_COUNT)
+        {
+            Thread.sleep(200);
```

Comment:
Probably should print the number of messages acked, and how many more you are waiting for.